lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -255,6 +256,47 @@ def add_embed_frameworks_script_phase
           end
         end
 
+        # Updates all target script phases fro the current target, including creating or updating, deleting
+        # and re-ordering.
+        #
+        # @return [void]
+        #
+        def update_target_script_phases
+          target_definition_script_phases = target.target_definition.script_phases
+          target_definition_script_phase_names = target_definition_script_phases.map { |k| k[:name] }
+          standard_script_phase_names = [CHECK_MANIFEST_PHASE_NAME, EMBED_FRAMEWORK_PHASE_NAME, COPY_PODS_RESOURCES_PHASE_NAME]
+          native_targets.each do |native_target|
+            # Delete script phases no longer present in the target definition.
```

Comment:
Also I'd be glad to review if you want to take a stab at it! It all starts in the DSL of cocoapods/core project :)